lib/hdprivatekey.js

```diff
@@ -127,6 +127,11 @@ HDPrivateKey._getDerivationIndexes = function(path) {
   return _.any(indexes, isNaN) ? null : indexes;
 };
 
+HDPrivateKey.prototype.derive = function() {
+  throw new Error('Method has been deprecated, please use deriveChild or deriveNonCompliantChild' +
+                  'and see documentation for more information');
```

Comment:
I unterstand that. So just completely remove the method. This will also throw an error of someone wants to use it, but then it does not show in IDE autocomplete and so on. Just think how annoying this method will be in the future.